### PR TITLE
Disable relevant parts of Panache when Hibernate ORM/Reactive is disabled

### LINKED ISSUE #28842: `quarkus.hibernate-orm.enabled = false` doesn't work when using Panache
### Describe the bug

If we try to set `quarkus.hibernate-orm-enabled = false` in an application that uses Panache, we get this error when building:

```
[ERROR] Failed to execute goal io.quarkus.platform:quarkus-maven-plugin:2.13.2.Final:build (default) on project managed-server-ui-cli: Failed to build quarkus application: io.quarkus.builder.ChainBuildException: No producers for required item class io.quarkus.hibernate.orm.deployment.JpaModelBuildItem -> [Help 1]
```

That's probably caused by `PanacheJpaCommonResourceProcessor` assuming that `JpaModelBuildItem` is always present; it should use `Optional<JpaModelBuildItem>` instead. Or it should just disable itself when Hibernate ORM is disabled.

And we should have tests for that...

### Expected behavior

_No response_

### Actual behavior

_No response_

### How to Reproduce?

_No response_

### Output of `uname -a` or `ver`

_No response_

### Output of `java -version`

_No response_

### GraalVM version (if different from Java)

_No response_

### Quarkus version or git rev

_No response_

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

_No response_

### Additional information

_No response_

### PR DESCRIPTION
**Pull request by yrodiere**
Fixes #28842

### CODE REVIEW

File: extensions/panache/hibernate-orm-panache/deployment/src/main/java/io/quarkus/hibernate/orm/panache/deployment/PanacheHibernateResourceProcessor.java [hunk 1]
```diff
@@ -58,11 +56,6 @@ public final class PanacheHibernateResourceProcessor {
 
     private static final DotName DOTNAME_ID = DotName.createSimple(Id.class.getName());
 
-    @BuildStep
```
Comment (geoand) [inline extensions/panache/hibernate-orm-panache/deployment/src/main/java/io/quarkus/hibernate/orm/panache/deployment/PanacheHibernateResourceProcessor.java:61]: What's the reason for removing this?

Comment (yrodiere) [reply extensions/panache/hibernate-orm-panache/deployment/src/main/java/io/quarkus/hibernate/orm/panache/deployment/PanacheHibernateResourceProcessor.java:61]: It's a mistake. Thanks for the heads-up!

Comment (geoand) [reply extensions/panache/hibernate-orm-panache/deployment/src/main/java/io/quarkus/hibernate/orm/panache/deployment/PanacheHibernateResourceProcessor.java:61]: 👌🏼

Comment (yrodiere) [reply extensions/panache/hibernate-orm-panache/deployment/src/main/java/io/quarkus/hibernate/orm/panache/deployment/PanacheHibernateResourceProcessor.java]: Fixed

**Review by geoand — APPROVED** [review]

**Comment by geoand** [comment]
Do we need a review from anyone else for this?

I personally think it's pretty safe...

**Comment by yrodiere** [comment]
> Do we need a review from anyone else for this?
> 
> I personally think it's pretty safe...

That makes two of us, so I'll merge. Thanks!

**Comment by geoand** [comment]
🆗

**Pull request merged**